src/app/config/no-config/no-config.component.ts

```diff
@@ -184,16 +218,41 @@ export class NoConfigComponent implements OnInit {
     this.router.navigate(['/main-screen']);
   }
 
-  public increasePage(): void {
+  public async increasePage(): Promise<any> {
     this.page += 1;
+    if (this.page <= 2) {
+      if (JSON.stringify(this.opInstance) != JSON.stringify(this.octoprintNodes['other'])) {
+        this.config.octoprint.url = 'http://' + this.opInstance['url'];
+        this.manualEntry = false;
+      } else {
+        this.config.octoprint.url = 'http://localhost:80/api/';
+        this.manualEntry = true;
+      }
+    } else if (this.config.octoprint.accessToken == '' && this.page > 2) {
+      this.page = 2;
+    } else if (this.page > 2) {
+      if (this.octoprintConnection === false) {
+        await this.testOctoprintAPI().then(res => {
+          this.opApiMsg = '';
+          if (this.opInstance.name != 'other') {
+            this.config.printer.name = this.opInstance.name;
+          } else {
+            this.config.printer.name = '';
+          }
+        }, err => {
+          this.opApiMsg = 'API Error: ' + this.octoprintConnectionError;
+          this.page = 2;
+        });
+      }
+    }
     if (this.page === this.totalPages) {
       this.createConfig();
     }
     this.changeProgress();
   }
 
   public decreasePage(): void {
-    if (this.page === 5) {
+    if (this.page == this.totalPages - 1) {
```

Comment:
👍

Reply:
Did I miss this one?

Reply:
No that was just an: I like that reaction, sorry for the confusion :)